runtime/lua/vim/filetype/detect.lua

```diff
@@ -619,6 +619,29 @@ function M.hw(_, bufnr)
   return 'virata'
 end
 
+-- This function checks for an assembly comment or a SWIG keyword or verbatim
+-- block in the first 50 lines.
+-- If not found, assume Progress.
+--- @type vim.filetype.mapfn
+function M.i(path, bufnr)
+  if vim.g.filetype_i then
+    return vim.g.filetype_i
+  end
+
+  local ft_swig_keywords =
+    [[^\s*%\%(addmethods\|apply\|beginfile\|clear\|constant\|define\|echo\|enddef\|endoffile\|extend\|feature\|fragment\|ignore\|import\|importfile\|include\|includefile\|inline\|insert\|keyword\|module\|name\|namewarn\|native\|newobject\|parms\|pragma\|rename\|template\|typedef\|typemap\|types\|varargs\|warn\)]]
+  local ft_swig_verbatim_block_start = [[^\s*%{]]
```

Comment:
```suggestion
  -- These include the leading '%' sign
  local ft_swig_keywords =
    [[^\s*%\%(addmethods\|apply\|beginfile\|clear\|constant\|define\|echo\|enddef\|endoffile\|extend\|feature\|fragment\|ignore\|import\|importfile\|include\|includefile\|inline\|insert\|keyword\|module\|name\|namewarn\|native\|newobject\|parms\|pragma\|rename\|template\|typedef\|typemap\|types\|varargs\|warn\)]]
  -- This is the start/end of a block that is copied literally to the processor file (C/C++)
  local ft_swig_verbatim_block_start = [[^\s*%{]]
```

In case you want to port the comments too. Also, I guess the `ft_swig_verbatim_block_start` pattern is simple enough to be a Lua pattern `^%s*%%{` if that's preferable. (I don't really know the recommended protocols for filetype.lua stuff :laughing:)